inferno-core/inferno-core.cabal

```diff
@@ -1,107 +1,114 @@
-bug-reports:         https://github.com/plow-technologies/inferno.git/issues
-copyright:           Plow-Technologies LLC
-license:             MIT
-license-file:        LICENSE
-author:              Sam Balco
-maintainer:          info@plowtech.net
-build-type:          Simple
+cabal-version:      2.4
+name:               inferno-core
+version:            0.11.6.0
+synopsis:           A statically-typed functional scripting language
+description:
+  Parser, type inference, and interpreter for a statically-typed functional scripting language
+
+category:           DSL,Scripting
+homepage:           https://github.com/plow-technologies/inferno.git#readme
+bug-reports:        https://github.com/plow-technologies/inferno.git/issues
+copyright:          Plow-Technologies LLC
+license:            MIT
+license-file:       LICENSE
+author:             Sam Balco
+maintainer:         info@plowtech.net
+build-type:         Simple
 extra-source-files:
-    CHANGELOG.md
-    golden/**/*.json
-    golden/**/*.bin
+  CHANGELOG.md
+  golden/**/*.bin
+  golden/**/*.json
 
 source-repository head
-  type: git
+  type:     git
   location: https://github.com/plow-technologies/inferno.git
 
 library
   exposed-modules:
-      Inferno.Core
-    , Inferno.Docs
-    , Inferno.Eval
-    , Inferno.Eval.Error
-    , Inferno.Infer
-    , Inferno.Infer.Env
-    , Inferno.Infer.Exhaustiveness
-    , Inferno.Infer.Pinned
-    , Inferno.Module
-    , Inferno.Module.Cast
-    , Inferno.Module.Builtin
-    , Inferno.Module.Prelude
-    , Inferno.Parse
-    , Inferno.Parse.Commented
-    , Inferno.Parse.Error
-    , Inferno.Instances.Arbitrary
-    , Inferno.Utils.QQ.Script
-    , Inferno.Utils.QQ.Module
+    Inferno.Core
+    Inferno.Docs
+    Inferno.Eval
+    Inferno.Eval.Error
+    Inferno.Infer
+    Inferno.Infer.Env
+    Inferno.Infer.Exhaustiveness
+    Inferno.Infer.Pinned
+    Inferno.Instances.Arbitrary
+    Inferno.Module
+    Inferno.Module.Builtin
+    Inferno.Module.Cast
+    Inferno.Module.Prelude
+    Inferno.Parse
+    Inferno.Parse.Commented
+    Inferno.Parse.Error
+    Inferno.Utils.QQ.Module
+    Inferno.Utils.QQ.Script
+
   other-modules:
-      Inferno.Infer.Error
-    , Inferno.Module.Prelude.Defs
-    , Inferno.Utils.QQ.Common
-  hs-source-dirs:
-      src
-  ghc-options: -Wall -Wunused-packages -Wincomplete-uni-patterns -Wincomplete-record-updates -Wincomplete-patterns
+    Inferno.Infer.Error
+    Inferno.Module.Prelude.Defs
+    Inferno.Utils.QQ.Common
+
+  hs-source-dirs:     src
+  ghc-options:
+    -Wall -Wunused-packages -Wincomplete-uni-patterns
+    -Wincomplete-record-updates -Wincomplete-patterns
+
   build-depends:
-      base                      >= 4.13 && < 4.17
-    , bimap                     >= 0.5.0 && < 0.6
-    , bytestring                >= 0.10.10 && < 0.12
-    , containers                >= 0.6.2 && < 0.7
-    , cryptonite                >= 0.30 && < 0.31
+    , base
```

Comment:
I decided to get rid of the strict version bounds, again for several reasons:
- We use Nix for this and don't need to involve the Cabal bounds solver again
- This is our practice elsewhere
- We added these version bounds only to appease `cabal sdist` for release to Hackage, but we haven't done a Hackage release in ~2 years. I don't think we even should, since a lot of the project would be very difficult to build outside of Nix
- Having these constraints makes upgrading harder
- `cabal gen-bounds` doesn't help very much